### ISSUE TITLE
How is RTP handled in heplify-server?

### ISSUE THREAD

**Issue by systemcrash** [sipcapture/heplify-server#343]
I turned on debug logging, and all I see currently looks like:

```
heplify-server    | 2020/03/11 13:07:42.341330 decoder.go:149: DBG [hep] &HEP{Version:2,Protocol:17,SrcIP:10.xx.xx.xx,DstIP:10.xx.xx.xx,SrcPort:64682,DstPort:58042,Tsec:1583932062,Tmsec:340607,ProtoType:4,NodeID:101,NodePW:asdf,Payload:	h`7$x{__ݰ޷߹pozruyy{^X~Zwܷ~zsqwvxux|^x[~tz,CID:102649YWNlMjdiNTYyMjgyMmM3Yjg2NGM1OTE3MjRlYjk3ZTc,Vlan:0,}
heplify-server    |
heplify-server    | 2020/03/11 13:07:42.356573 decoder.go:149: DBG [hep] &HEP{Version:2,Protocol:17,SrcIP:10.xx.xx.xx,DstIP:10.xx.xx.xx,SrcPort:7078,DstPort:58040,Tsec:1583932062,Tmsec:352607,ProtoType:4,NodeID:101,NodePW:asdf,Payload:
heplify-server    | ФC_ypwߝw{_ݘywvw{ڸww_]Z_۶x]yyxzv^^߻ݟ,CID:102649YWNlMjdiNTYyMjgyMmM3Yjg2NGM1OTE3MjRlYjk3ZTc,Vlan:0,}
heplify-server    |
heplify-server    | 2020/03/11 13:07:42.360377 decoder.go:149: DBG [hep] &HEP{Version:2,Protocol:17,SrcIP:10.xx.xx.xx,DstIP:10.xx.xx.xx,SrcPort:64682,DstPort:58042,Tsec:1583932062,Tmsec:356608,ProtoType:4,NodeID:101,NodePW:asdf,Payload:	i7$xܟxWܻV\sxuw|nt|xu{xwWvUux~\|u~r~|wwt|:,CID:102649YWNlMjdiNTYyMjgyMmM3Yjg2NGM1OTE3MjRlYjk3ZTc,Vlan:0,}
heplify-server    |
heplify-server    | 2020/03/11 13:07:42.366697 decoder.go:149: DBG [hep] &HEP{Version:2,Protocol:17,SrcIP:10.xx.xx.xx,DstIP:10.xx.xx.xx,SrcPort:7078,DstPort:58040,Tsec:1583932062,Tmsec:364608,ProtoType:4,NodeID:101,NodePW:asdf,Payload:
heplify-server    | ФC7_u_޺{߳vussx^ݻwyo_ww{w_zߺu9v]{y],CID:102649YWNlMjdiNTYyMjgyMmM3Yjg2NGM1OTE3MjRlYjk3ZTc,Vlan:0,}
```

I see that Payload logging could be improved with a (additional) type check. e.g. If payload isn't SIP or JSON-RTCP, truncate or don't display (raw) Payload.

**Comment by lmangani** [comment]
Hi paul! Thanks for getting started on this, but I think some preparation would benefit your efforts. First and foremost - RTP is not handled in Heplify-Server and we do not mirror RTP across using HEP. This is not because of lack of will or capabilities, but due to specific decisions - I think you're missing out on the design of our platform, which is heavily against privacy invasion (recording and spying) and heavily pro synthesized data in consumable formats that does not create GDPR compliance issues for our users. 

I would suggest participating one of our dev meetings or outlining your proposal. Introduce yourself to the group, don't be a stranger, and we'll make things happen easily.

TLDR; it is not. If this is an attempted change request, it will need coordination ;)

**Comment by systemcrash** [comment]
Hi! OK, so *currently* you don't handle RTP. I understand your intentions - but the very concept of capture is already 'contra' privacy, I think. Capture of any kind raises privacy questions, although it's not the tool but the user who is responsible. 

I'm was trying to figure out what happens with RTP, not (yet) making a suggestion. I *think* that handling RTP has benefits in the VoIP scene, especially with regards to momentary trouble-shooting. This is an area I predominantly work within, and now and then, having eyes on RTP also is necessary. I don't intend for continuous RTP ingestion - mainly temporary. But having continuous ingestion also has benefits: heplify can make its own determinations on RTP metrics, in addition to relying on JSON interpretations of RTCP.

I'll happily participate in a dev meeting (I joined the dev list, but when is next one?).

**Comment by negbie** [comment]
@systemcrash what you could do is to use heplify with -wf and -zf flags to write pcaps to disk. These will include RTP and you can use tools like google/stenographer to look into them at scale.

**Comment by systemcrash** [comment]
OK. Good tip. Where would I configure those flags in the yaml (docker)?

**Comment by negbie** [comment]
This is for heplify client and you configure them like usual with cli flags.

**Comment by systemcrash** [comment]
Ah, OK - not an option. The HEP agent is running on a closed system. Nice to know these things exist though.

**Comment by lmangani** [comment]
@systemcrash you can add any command lines to the container it should be quite simple - this said, if you consistently need all data, then you might just want to run stenographer as a rotating cache (docker or else) and and use [https://github.com/QXIP/node-webshark] webshark and/or ste steonographer API from Homer using HepSub (which already exists somewhere)

**Comment by systemcrash** [comment]
Hmm, those moving parts will probably take a while to figure out. But if stenographer ingests tons of packets, HepSub can differentiate based on Call-IDs? Call-IDs+RTP SSRC are associated? This is what is most valuable.

**Comment by lmangani** [comment]
Closing as non issue, but happy to continue the discussion about this topic further in the thread.

**Comment by lmangani** [comment]
@systemcrash the truth is the best method depends on what you're after. since we don't know what your ultimate goal is, its hard to tell but there are plenty of answers possible.
